code/sorting/bubble_sort/efficientbubblesort.cpp

```diff
@@ -0,0 +1,53 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+void bubbleSort(vector<int> &v){
+	int size = v.size();
+	
+	for(int i=0; i<size-1; i++){
+	
+		bool isSorted=1;
+	
+		for(int j=0; j<size-1-i; j++){
+			if(v[j]>v[j+1]){
+				swap(v[j],v[j+1]);
+				isSorted = 0;
+			}
+		}
+	
+		for(int j=size-2; j>=i; j--){
+			if(v[j]>v[j+1]){
+				swap(v[j],v[j+1]);
+				isSorted = 0;
+			}
+		}
+	
+		if(isSorted) break;
+	
+	}
+	/*
+	In this implementation,
+	Modification From Normal Bubble Sort:
+	1) Sorting From Both the Sides Simultaneously
+	2) A variable isSorted which automatically breaks out of loop when elements get sorted reducing the total number of iterations.
+	*/
+}
+
+int main(){
+	int n;
+	cout<<"Enter the number of elements you want to enter: "<<endl;
+	cin>>n;
```

Comment:
When using vectors, you should allow the user to enter unlimited input without having to specify the size. (This is where `vector::push_back` would be useful).

Reply:
So how will I detect the end of input, by newline character?

Reply:
@arnavb can you please tell me what should I read ahead or add over here, I have knowledge of linked lists, queues, trees, stacks, priority queues, heap (binomial, ternary, Fibonacci), and Graph(just BFS and DFS).

Reply:
@apurva91 You could use the newline character, but the conventional way to do it is until an EOL/EOF.